python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -909,6 +909,7 @@ def assembleMetadata(self, coaddExposure, tempExpRefList, weightList):
             self.shrinkValidPolygons(coaddInputs)
 
         coaddInputs.visits.sort()
+        coaddInputs.ccds.sort()
```

Comment:
On a plane and can't really see for myself, what does this do?

Reply:
So, when you look at a `coaddInputs` object, you see, e.g.
```
In [7]: coaddInputs.visits[0:3]
Out[7]: 
<class 'lsst.afw.table.ExposureCatalog'>
 id  bbox_min_x bbox_min_y bbox_max_x bbox_max_y goodpix        weight       filter
        pix        pix        pix        pix                                       
---- ---------- ---------- ---------- ---------- -------- ------------------ ------
1228      15900      15900      20099      20099 15689141  116.7270915916109  HSC-I
1230      15900      15900      20099      20099 16580225 116.36207676312476  HSC-I
1232      15900      15900      20099      20099 16589337 118.26125052140408  HSC-I
```
These all happen to be in order, but if they weren't, the call to `coaddInputs.visits.sort()` orders that list by the `id` column, which happens to == `visit` in this case.  If you further look at the `.ccds` attribute, you see, e.g.
```
In [8]: coaddInputs.ccds[:9]
Out[8]: 
<class 'lsst.afw.table.ExposureCatalog'>
  id   bbox_min_x bbox_min_y bbox_max_x bbox_max_y ccd visit goodpix       weight       filter
          pix        pix        pix        pix                                                
------ ---------- ---------- ---------- ---------- --- ----- ------- ------------------ ------
245640          0          0       2047       4175  40  1228  226463  116.7270915916109  HSC-I
245641          0          0       2047       4175  41  1228 1072977  116.7270915916109  HSC-I
245648          0          0       2047       4175  48  1228 1368500  116.7270915916109  HSC-I
245649          0          0       2047       4175  49  1228 6589704  116.7270915916109  HSC-I
245656          0          0       2047       4175  56  1228 1114850  116.7270915916109  HSC-I
245657          0          0       2047       4175  57  1228 5316647  116.7270915916109  HSC-I
246049          0          0       2047       4175  49  1230 6462585 116.36207676312476  HSC-I
246057          0          0       2047       4175  57  1230 8404254 116.36207676312476  HSC-I
246065          0          0       2047       4175  65  1230 1713386 116.36207676312476  HSC-I
```
The `id` column here is "some big number + ccd (e.g., for visit 1228 it is 245600 + ccdId).  So sorting on the `id` here effectively sorts on visit (big number) and sub-sorts on the detecter per visit.  An example, if I change the first entry to be ccd 99:
```
In [9]: coaddInputs.ccds[0]["id"] = 245699
In [10]: coaddInputs.ccds[0]["ccd"] = 99
In [11]: coaddInputs.ccds[:9]
Out[11]: 
<class 'lsst.afw.table.ExposureCatalog'>
  id   bbox_min_x bbox_min_y bbox_max_x bbox_max_y ccd visit goodpix       weight       filter
          pix        pix        pix        pix                                                
------ ---------- ---------- ---------- ---------- --- ----- ------- ------------------ ------
245699          0          0       2047       4175  99  1228  226463  116.7270915916109  HSC-I
245641          0          0       2047       4175  41  1228 1072977  116.7270915916109  HSC-I
245648          0          0       2047       4175  48  1228 1368500  116.7270915916109  HSC-I
245649          0          0       2047       4175  49  1228 6589704  116.7270915916109  HSC-I
245656          0          0       2047       4175  56  1228 1114850  116.7270915916109  HSC-I
245657          0          0       2047       4175  57  1228 5316647  116.7270915916109  HSC-I
246049          0          0       2047       4175  49  1230 6462585 116.36207676312476  HSC-I
246057          0          0       2047       4175  57  1230 8404254 116.36207676312476  HSC-I
246065          0          0       2047       4175  65  1230 1713386 116.36207676312476  HSC-I

In [12]: coaddInputs.ccds.sort()
In [13]: coaddInputs.ccds = coaddInputs.ccds.copy(deep=True)

In [18]: coaddInputs[:9]
Out[18]: 
<class 'lsst.afw.table.ExposureCatalog'>
  id   bbox_min_x bbox_min_y bbox_max_x bbox_max_y ccd visit goodpix       weight       filter
          pix        pix        pix        pix                                                
------ ---------- ---------- ---------- ---------- --- ----- ------- ------------------ ------
245641          0          0       2047       4175  41  1228 1072977  116.7270915916109  HSC-I
245648          0          0       2047       4175  48  1228 1368500  116.7270915916109  HSC-I
245649          0          0       2047       4175  49  1228 6589704  116.7270915916109  HSC-I
245656          0          0       2047       4175  56  1228 1114850  116.7270915916109  HSC-I
245657          0          0       2047       4175  57  1228 5316647  116.7270915916109  HSC-I
245699          0          0       2047       4175  99  1228  226463  116.7270915916109  HSC-I
246049          0          0       2047       4175  49  1230 6462585 116.36207676312476  HSC-I
246057          0          0       2047       4175  57  1230 8404254 116.36207676312476  HSC-I
246065          0          0       2047       4175  65  1230 1713386 116.36207676312476  HSC-I
```
And you see that ccd the ordering is now increasing for visit 1228 (without changing 1230, which was already in order).  In case you were wondering, the `sort()` leaves them Non-contiguous, hence the deep copy.

Let me know if this is still unclear!

I don't have a handy "how-to-reproduce", but, yes this DID happen 😉